File: Assets/Scripts/PlayerMovement.cs
using UnityEngine;
using System.Collections;

public class PlayerMovement : MonoBehaviour {

	public float thrust = 300.0f;
	public float rotationSpeed = 40.0f;
	private float accel = 0;
	private float rotation = 0;
	private float radRotation = 0;

	// Use this for initialization
	void Start () {
	
	}
	
	// Update is called once per frame
	void FixedUpdate () {

	    accel = Input.GetAxis("Vertical") * thrust;
        rotation = Input.GetAxis("Horizontal") * -(rotationSpeed);
	    radRotation = (rigidbody2D.rotation * (Mathf.PI / 180));

	    rigidbody2D.AddRelativeForce(Vector2.up * accel);
	    rigidbody2D.MoveRotation(rigidbody2D.rotation + rotation * Time.deltaTime);	
	
	    if(Input.GetButton("Fire1")){
			Vector2 forceVec = new Vector2(-rigidbody2D.velocity.normalized.x * thrust, -rigidbody2D.velocity.normalized.y * thrust);
		    rigidbody2D.AddForce(forceVec,ForceMode2D.Force);
	    };

		if(Input.GetKey (KeyCode.Q)){
			rigidbody2D.AddRelativeForce(Vector2.right * -thrust);
		}

		if(Input.GetKey (KeyCode.E)){
			rigidbody2D.AddRelativeForce(Vector2.right * thrust);
		}
	}
}

File: Assets/Scripts/PlayerMovement.js
#pragma strict

var thrust : float = 5.0;
var rotationSpeed : float = 75.0;
public var accel : float = 0;
public var rotation : float = 0;
public var radRotation : float = 0;

//var velocity: Vector2;

function Start () {

}

function FixedUpdate () {
	accel = Input.GetAxis("Vertical") * thrust;
	rotation = Input.GetAxis("Horizontal") * -(rotationSpeed);
	radRotation = (rigidbody2D.rotation * (Mathf.PI / 180));
	
	var vx = -accel * (Mathf.Sin(radRotation));
	var vy = accel * (Mathf.Cos(radRotation));
	
	rigidbody2D.AddForce(Vector2(vx, vy));
	rigidbody2D.MoveRotation(rigidbody2D.rotation + rotation * Time.deltaTime);	
	
	if(Input.GetButton("Fire1")){
		var forceVec : Vector2 = -rigidbody2D.velocity.normalized * thrust;
		Debug.Log(forceVec);
		rigidbody2D.AddForce(forceVec,ForceMode2D.Force);
	};
}